sabnzbd/filesystem.py

```diff
@@ -944,6 +944,54 @@ def remove_all(path: str, pattern: str = "*", keep_folder: bool = False, recursi
 ##############################################################################
 # Diskfree
 ##############################################################################
+def disk_free_macos_clib_statfs64(directory):
+    # MacOS only!
+    # direct system call to c-lib's statfs(), not python's os.statvfs()
+    # because statvfs() on MacOS has a rollover at 4TB (possibly a 32bit rollover with 10bit block size)
+    # Based on code of pudquick and blackntan
+    # Input: directory.
+    # Output: disksize and available space, in bytes
+
+    import ctypes
+    import ctypes.util
+
+    # format & parameters: on MacOS, see "man statfs", lines starting at
+    # "struct statfs { /* when _DARWIN_FEATURE_64_BIT_INODE is defined */"
+    class statfs64(ctypes.Structure):
+        _fields_ = [
+            ("f_bsize", ctypes.c_uint32),
+            ("f_iosize", ctypes.c_int32),
+            ("f_blocks", ctypes.c_uint64),
+            ("f_bfree", ctypes.c_uint64),
+            ("f_bavail", ctypes.c_uint64),
+            ("f_files", ctypes.c_uint64),
+            ("f_ffree", ctypes.c_uint64),
+            ("f_fsid", ctypes.c_uint64),
+            ("f_owner", ctypes.c_uint32),
+            ("f_type", ctypes.c_uint32),
+            ("f_flags", ctypes.c_uint32),
+            ("f_fssubtype", ctypes.c_uint32),
+            ("f_fstypename", ctypes.c_char * 16),
+            ("f_mntonname", ctypes.c_char * 1024),
+            ("f_mntfromname", ctypes.c_char * 1024),
+            ("f_reserved", ctypes.c_uint32 * 8),
+        ]
+
+    fs_info64 = statfs64()  # set up the parameters to be filled out
+    result = sabnzbd.MACOSLIBC.statfs64(
+        ctypes.create_string_buffer(utob(directory)), ctypes.byref(fs_info64)
+    )  # fs_info64 gets filled out via the byref()
+    if result == 0:
+        # Upon successful completion, a value of 0 is returned.
+        disk_size = fs_info64.f_blocks * fs_info64.f_bsize
+        avail_size = fs_info64.f_bavail * fs_info64.f_bsize
+    else:
+        # Otherwise, -1 is returned and the global variable errno is set to indicate the error.
+        logging.debug("Call to MACOSLIBC.statfs64 not successful. Value of errno is %s", ctypes.get_errno())
+        disk_size = avail_size = 0
```

Comment:
Just `return 0.0, 0.0`

Reply:
I'll make it `return 0, 0` as it's bytes (like win32api.GetDiskFreeSpaceEx() ), not GIGIbytes (and thus float)